### ISSUE TITLE
Remove story poster image fallback

### ISSUE THREAD

**Issue by swissspidy** [GoogleForCreators/web-stories-wp#3671]
## Feature Description

When users don't add a story cover image (poster image), we use the following image as a fallback:

https://github.com/google/web-stories-wp/blob/7fc1c9af6a1e43c0edc2ec57a45f82eccb1d8d08/assets/images/fallback-poster.jpg

This is basically a last resort to ensure the image meets the poster requirements.

**UPDATE 9/10/2020:** We received feedback that setting a fallback cover image on behalf of the user is not a good idea. We should, for now, NOT add a fallback image at all, and encourage users to set a meaningful cover image **[following the guidelines](https://amp.dev/documentation/components/amp-story/#poster-guidelines-(for-poster-portrait-src,-poster-landscape-src,-and-poster-square-src))**

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

**Comment by bmattb** [comment]
This was broken up to address the two issues:
1. BUG - fallback image is not the right size - https://github.com/google/web-stories-wp/issues/3839
2. fallback is bad image and can be improved - this ticket

**Comment by bmattb** [comment]
@o-fernandez please confirm if this ticket (regarding improvment) is needed for stable release

**Comment by spacedmonkey** [comment]
An idea discussed in sprint planning, was if a user do not select a poster image, we generate one for you. Similar to how video poster image works currently. Basically take a snapshot of the first page of the story and save it as an image, then set it as the poster image. 

Not sure what this should look like for the user. Maybe a nag in the checklist. 
```
"It appears that you did not upload an a poster image. Poster images are requires"
<button>Upload custom poster</button> 
<button>Generate from first page of story</button>
```

Or if it should just do this for you automatically if a poster is not set. 

I would love feedback from @o-fernandez and @samitron7 on this idea, as is a product spec change and needs UX.

**Comment by swissspidy** [comment]
^ Above would be in-line with [guidelines](https://amp.dev/documentation/components/amp-story/?referrer=ampproject.org#poster-guidelines-(for-poster-portrait-src,-poster-landscape-src,-and-poster-square-src)) too:

> * The poster image should be representative of the entire AMP story.
> * The poster image should be visible to the user when the AMP story begins.

**Comment by samitron7** [comment]
The poster image is the first thing viewers sees on any surface so we should probably not even allow publishing without it. A story without a cover looks very broken and unprofessional for the creator and the surface it's represented on.

**Comment by swissspidy** [comment]
Nobody is saying otherwise.

But instead of having the user open Photoshop to create a cover image that follows best practices, or having them take a screenshot of the first page that also meets the guidelines, we can just create a poster for them.

**Comment by samitron7** [comment]
Yes, was just replying to the alternative consideration "Should we make the poster image mandatory before publishing?". Was a bit confused by that.

@spacedmonkey Let's automatically generate one for the users (+1 to your recommendations here) and they can change if they like.  We can flag as guidance in the pre-publish list and I'll need to think about whether or not we need additional buttons with the message. This will depend on how I build the pre-publish list.

**Comment by spacedmonkey** [comment]
I think we should close this ticket and create a new ticket, that is assigned to the workspace pod. 

My recommendation would be if a user tries to publish the story without a poster only then it generates this poster.

**Comment by o-fernandez** [comment]
Yes, we should definitely do this. @swissspidy and @spacedmonkey is this something you can work on without a product brief/spec, or would you wait until we can provide one? LMK and I'll try to prioritize as necessary.

**Comment by swissspidy** [comment]
@o-fernandez Yeah this can be easily added behind a flag without requiring a brief or UX. Just noting that it's more fitting for the workspace pod' as it's not really a WordPress-related feature. Happy to consult on it though, or plan it for one of our next sprints if we do have some capacity left.

**Comment by o-fernandez** [comment]
Makes sense - let me just put it under ready for sprint and we can all collaborate on the solution once it's being picked up. This is pretty important for launch, so will keep as a 1.0 Stable requirement for now.

**Comment by o-fernandez** [comment]
This was flagged as a potential 1.0 launch blocker, let's discuss if there's a quick solution for it. Adding back to 1.0 Stable scope.

**Comment by bmattb** [comment]
@carlos-kelly to connect with @swissspidy and confirm on how to address this.

**Comment by barklund** [comment]
@carlos-kelly, can you make sure to link the PR for this one? I can't (easily) find it.

**Comment by swissspidy** [comment]
@barklund Linked PR #4350 correctly 👍

**Comment by o-fernandez** [comment]
This needs more work. Currently, it's not generating a good cover image at all. Let's consider options and get some workable solution in for Stable. 

In addition, we should ensure that the cover image is autogenerated upon publishing if there is no cover image set. Right now, I don't see that happening.

**Comment by o-fernandez** [comment]
Made non-blocking for RC1, even though it's a shame not to have this working. Will start a thread to get something in for the Stable release.

**Comment by o-fernandez** [comment]
Thoughts to unblock stable -- let's make this a very simple solution. As mentioned above: 
1. Best case scenario, the user sets a cover image themselves that follows our [cover image guidelines](https://amp.dev/documentation/components/amp-story/#poster-guidelines-(for-poster-portrait-src,-poster-landscape-src,-and-poster-square-src)). 
2. If the user doesn't set a cover image, we **autogenerate one and set it automatically at publish time.** 

Generation options: 
1. If the first page of the story has a bg image or video, use an appropriately cropped version of that as the poster/cover image. 
2. If the first page doesn't have a bg image/video, find the first image/video in the story (Starting in page 1) and use that instead. In the case that there are multiple media elements on the first page with media, use the "largest" one based on area it takes on that page in the canvas. 

EXCEPTION: if the image/media resolution is low enough that the cover image would not meet minimum dimensions (at least 640 x 853) we should skip that and find another one. 

This is not a great solution, but it's better than the current situation in which there's no poster/cover image set at all.

**Comment by samitron7** [comment]
Yeah that looks right to me @o-fernandez. In the case where the story has no image/video, can we generate a default? This is better than no image.

**Comment by swissspidy** [comment]
> In the case where the story has no image/video, can we generate a default? This is better than no image.

Our very last resort in case there is absolutely no poster is currently this one:

<img width="320" alt="fallback-poster" src="https://user-images.githubusercontent.com/841956/92777123-2c31ce80-f3a0-11ea-86b1-fbc03d1f7ce0.png">

**Comment by dreamofabear** [comment]
Feedback from weekly review: prefer block publishing vs. using a placeholder poster (in the last resort case where the story has no images/videos).

Since we want to avoid blocking publishing due to story guidelines in general (e.g. missing title), let's actually remove the fallback image and let the story have no poster (become invalid). This will avoid ugly stories showing up on surfaces we care about.

Ditto re: missing title.

**Comment by swissspidy** [comment]
@choumx Is this specifically referring to https://github.com/google/web-stories-wp/issues/3671#issuecomment-690584619 above? If there are some images we could use for the poster, we should try to use that still, correct?

Edit: saw the clarification now

**Comment by swissspidy** [comment]
We might want to split this ticket up:

- One ticket for removing the existing fallback (use this one?)
- One ticket for encouraging users to add a cover image (new one)

**Comment by o-fernandez** [comment]
Yes, this ticket is just for "remove fallback", don't use a fallback any longer. 

For encouraging users to add a cover image, that should be covered by #78, so hopefully we prioritize that.

**Comment by bmattb** [comment]
We will need detailed testing instructions for this as it's quite complex.

**Comment by swissspidy** [comment]
> We will need detailed testing instructions for this as it's quite complex.

It's actually quite simple to test 🙂 

Testing instructions added to #4480

**Comment by westonruter** [comment]
FYI: AMP plugin [v2.0.2](https://github.com/ampproject/amp-wp/releases/tag/2.0.2) has been released which facilitates Web Stories being served as without the `poster-portrait-src` while also omitting the `amp` attribute to prevent GSC from complaining about known invalid AMP documents. For relevant change, see https://github.com/ampproject/amp-wp/pull/5388.
